### PR TITLE
Add tests for `Intl.PluralRules.prototype.resolvedOptions().pluralCategories` order

### LINKED ISSUE #4261: Need to add test for ECMA402 PR918
https://github.com/tc39/ecma402/pull/918 merged after reach consense in Oct 2024 TC39 meeting. We need to add test to verify the implementation

@ben-allen @sffc @anba

### PR DESCRIPTION
**Pull request by ben-allen**
Array elements should appear in following order: "zero", "one", "two", "few", "many", "other"

see https://github.com/tc39/ecma402/pull/918/

fix #4261

### CODE REVIEW

File: test/intl402/PluralRules/prototype/resolvedOptions/plural-categories-order.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+assert.compareArray(new Intl.PluralRules('en').resolvedOptions().pluralCategories, ['one', 'other'], "pluralCategories order or contents incorrect for 'en' locale");
+assert.compareArray(new Intl.PluralRules('fa').resolvedOptions().pluralCategories, ['one', 'other'], "pluralCategories order or contents incorrect for 'fa' locale");
+assert.compareArray(new Intl.PluralRules('fr').resolvedOptions().pluralCategories, ['one', 'many', 'other'], "pluralCategories order or contents incorrect for 'fr' locale");
+assert.compareArray(new Intl.PluralRules('sl').resolvedOptions().pluralCategories, ['one', 'two', 'few', 'other'], "pluralCategories order or contents incorrect for 'sl' locale");
+assert.compareArray(new Intl.PluralRules('gv').resolvedOptions().pluralCategories, ['one', 'two', 'few', 'many', 'other'], "pluralCategories order or contents incorrect for 'gv' locale");
+assert.compareArray(new Intl.PluralRules('ar').resolvedOptions().pluralCategories, ['zero', 'one', 'two', 'few', 'many', 'other'], "pluralCategories order or contents incorrect for 'ar' locale");
+
+
+
```
[the remark this answers is not in the document]
Comment (ben-allen) [reply test/intl402/PluralRules/prototype/resolvedOptions/plural-categories-order.js:23]: d'oh. fixed!

File: test/intl402/PluralRules/prototype/resolvedOptions/plural-categories-order.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+esid: sec-intl.pluralrules.prototype.resolvedoptions
+description: >
+  Tests that Intl.PluralRules.prototype.resolvedOptions elements given in correct order.
+info: |
+  Intl.PluralRules.prototype.resolvedOptions ()
+
+  4. Let pluralCategories be a List of Strings containing all possible results of PluralRuleSelect for the selected locale pr.[[Locale]], sorted according to the following order: "zero", "one", "two", "few", "many", "other".
+
+includes: [compareArray.js]
+---*/
```
[the remark this answers is not in the document]
Comment (ben-allen) [reply test/intl402/PluralRules/prototype/resolvedOptions/plural-categories-order.js]: fixed! (also alphabetized the locales just to be doing it)

File: test/intl402/PluralRules/prototype/resolvedOptions/plural-categories-order.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ms2ger) [reply test/intl402/PluralRules/prototype/resolvedOptions/plural-categories-order.js:23]: You seem to have missed the one between ko and sl, I'm afraid

[the remark this answers is not in the document]
Comment (ben-allen) [reply test/intl402/PluralRules/prototype/resolvedOptions/plural-categories-order.js]: Sigh. Sorry about that, should be fixed now

**Review by anba — APPROVED** [review]
Expected results match SM with <https://bugzilla.mozilla.org/show_bug.cgi?id=1930465> applied.

**Review by Ms2ger — APPROVED** [review]
Thanks!

**Pull request merged**